src/libs/Navigation/AppNavigator/useNavigationResetRootOnLayoutChange.ts

```diff
@@ -0,0 +1,16 @@
+import {useEffect} from 'react';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import navigationRef from '@libs/Navigation/navigationRef';
+
+function useNavigationResetRootOnLayoutChange() {
```

Comment:
Can we add a comment description of what this is used for.

Reply:
Would be good to start setting this up for more unit tests in the navigation are too @WojtekBoman @adamgrzybowski